synthtool/metadata.py

```diff
@@ -202,10 +214,27 @@ def __exit__(self, type, value, traceback):
             _add_new_files(tracked_new_files)
             _remove_obsolete_files(self.old_metadata)
         _append_git_logs(self.old_metadata, get())
-        _clear_local_paths(get())
+        if should_combine_commit_logs():
+            _combine_commit_logs()
+        _clear_local_paths()
         write(self.metadata_file_path)
 
 
+def _get_commit_log_since(path, sha):
+    """Invokes git to get the commit log in the given path since the given sha.
+
+    Returns:
+        string, the git log.
+    """
+    git = shutil.which("git")
+    output = subprocess.run(
+        [git, "-C", path, "log", "--pretty=%H%n%B", "--no-decorate", f"{sha}..HEAD"],
```

Comment:
For `googleapis/googleapis` this ends up being a fairly large list of updates, because it will include every update to the repo, what if we were clever and filtered the PRs to a sub-directory in `googleapis`, based on the API name?

I think we could potentially piggyback off the variables being passed in to the generator like `proto_path`.

Reply:
alternatively, if piggybacking is a pain, perhaps we could populate an additional variable in `synth.py`.